agimus_controller_ros/agimus_controller_ros/agimus_controller.py

```diff
@@ -0,0 +1,293 @@
+from agimus_controller_ros.parameters import AgimusControllerNodeParameters
+from agimus_controller.trajectory import (
+    TrajectoryBuffer,
+    TrajectoryPoint,
+    TrajectoryPointWeights,
+    WeightedTrajectoryPoint,
+)
+
+
+class AgimusController(Node):
+    def __init__(
+        self,
+        node_name: str = "agimus_controller_node",
+        params: AgimusControllerNodeParameters = None,
+    ) -> None:
+        super().__init__(node_name)
+        if params is None:
+            self.set_params()
+        else:
+            self.params = params
+        self.traj_buffer = TrajectoryBuffer()
+        self.last_point = None
+        self.first_run_done = False
+
+        self.initialize_ros_attributes()
+        self.get_logger().info("Init done")
+
+    def initialize_ros_attributes(self):
+        self.sensor_msg = None
+        self.control_msg = None
+        self.state_subscriber = self.create_subscription(
+            Sensor,
+            "sensor",
+            self.sensor_callback,
+            qos_profile=QoSProfile(
+                depth=10,
+                reliability=ReliabilityPolicy.BEST_EFFORT,
+            ),
+        )
+        self.subscriber_robot_description = self.create_subscription(
+            String,
+            "/robot_description",
+            self.robot_description_callback,
+            qos_profile=QoSProfile(
+                depth=1,
+                durability=DurabilityPolicy.TRANSIENT_LOCAL,
+                reliability=ReliabilityPolicy.RELIABLE,
+            ),
+        )
+        self.subscriber_mpc_input = self.create_subscription(
+            MpcInput,
+            "mpc_input",
+            self.mpc_input_callback,
+            qos_profile=10,
+        )
+        self.control_publisher = self.create_publisher(Control, "/control", 10)
+        self.ocp_solve_time_pub = self.create_publisher(Duration, "ocp_solve_time", 10)
+        self.ocp_x0_pub = self.create_publisher(Sensor, "ocp_x0", 10)
+        self.create_timer(self.params.ocp.dt, self.first_run_callback)
+        self.create_timer(self.params.ocp.dt, self.run_callback)
+        self.first_robot_sensor_msg_received = False
+        self.first_pose_ref_msg_received = True
+
+    def set_params(self):
+        """Setting the parameters of the node by reading a yaml file."""
+        self.declare_parameter("package_name", "")
+        package_name = (
+            self.get_parameter("package_name").get_parameter_value().string_value
+        )
+        self.declare_parameter("path_to_yaml", "")
+        path_to_yaml = (
+            self.get_parameter("path_to_yaml").get_parameter_value().string_value
+        )
+        yaml_path = Path(get_package_share_directory(package_name)) / path_to_yaml
+        with open(yaml_path, "r") as file:
+            params = yaml.safe_load(file)["agimus_controller_node"]["ros__parameters"]
+        self.params = AgimusControllerNodeParameters(params)
```

Comment:
I was doing this because I was hoping to reinstatiate the node class without ros to replay the simulation but it doesn't seem to be possible so I'm moving to the generate_parameter_library